apps/test/unit/templates/curriculumCatalog/CurriculumCatalogTest.jsx

```diff
@@ -121,4 +68,175 @@ describe('CurriculumCatalog', () => {
+    assert(noDeviceFilterCheckbox.checked);
+
+    // Filters for all courses compatible with chromebooks and tablets
+    const numFilteredCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numFilteredCurriculumCards).to.equal(
+      tabletAndNoDeviceShownCurricula.length
+    );
+  });
+
+  it('filtering by each filter shows subset of courses that match the filters', () => {
+    const numTotalCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numTotalCurriculumCards).to.equal(allShownCurricula.length);
+
+    // Select "Grade 2" and "Grade 3" in grade level filter
+    const grade2FilterCheckbox = screen.getByDisplayValue('grade_2');
+    fireEvent.click(grade2FilterCheckbox);
+    assert(grade2FilterCheckbox.checked);
+    const grade3FilterCheckbox = screen.getByDisplayValue('grade_3');
+    fireEvent.click(grade3FilterCheckbox);
+    assert(grade3FilterCheckbox.checked);
+
+    // Select "Physical Computing" and "Interdisciplinary" in topic filter
+    const physicalCompFilterCheckbox =
+      screen.getByDisplayValue('physical_computing');
+    fireEvent.click(physicalCompFilterCheckbox);
+    assert(physicalCompFilterCheckbox.checked);
+    const interdisciplinaryFilterCheckbox =
+      screen.getByDisplayValue('interdisciplinary');
+    fireEvent.click(interdisciplinaryFilterCheckbox);
+    assert(interdisciplinaryFilterCheckbox.checked);
+
+    // Select "Tablet" and "No Device" in device filter
+    const tabletFilterCheckbox = screen.getByDisplayValue('tablet');
+    fireEvent.click(tabletFilterCheckbox);
+    assert(tabletFilterCheckbox.checked);
+    const noDeviceFilterCheckbox = screen.getByDisplayValue('no_device');
+    fireEvent.click(noDeviceFilterCheckbox);
+    assert(noDeviceFilterCheckbox.checked);
+
+    // Filters for all courses that support:
+    // - Grades 2 or 3
+    // - Physical Computing or Interdisciplinary topics
+    // - Chromebooks or tablets
+    const numFilteredCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numFilteredCurriculumCards).to.equal(
+      multipleFiltersAppliedShownCurricula.length
+    );
+  });
+
+  it('applying every filter only filters out courses that have null for one of the filtered properties', () => {
+    const numTotalCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numTotalCurriculumCards).to.equal(allShownCurricula.length);
+
+    // Select all checkboxes
+    screen.getAllByRole('checkbox').forEach(checkbox => {
+      fireEvent.click(checkbox);
+      assert(checkbox.checked);
+    });
+
+    // With every filter applied
+    const numFilteredCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numFilteredCurriculumCards).to.equal(
+      allFiltersAppliedShownCurricula.length
+    );
+  });
```

Comment:
interesting, this seems like a good case to test for the frontend code, but also it seems worth following up to find out if we should be trying to prevent `null` values on these fields in the database so that we don't end up in this somewhat confusing situation. @dmcavoy am I correct in understanding that none of the filterable fields should be null in the database? if so, what do you think about making that an explicit requirement? if not, I am curious what an example is of a use case for nulls.

Reply:
I agree that anything that we are using to filter based on should not be null. I think Meg has been working on pointing out where this could happen and either preventing the card from showing or adding validations. Meg logged this ticket https://codedotorg.atlassian.net/browse/ACQ-557 for some of the fields. If there are other fields we should add to the list lets do that. Also if we need to pull this up and tackle it soon lets do that.

Reply:
Sounds good! So maybe once the data is validated on the backend then we can remove the redundant checks for non-null values on the front-end? Also as an additional note, I've noticed at least one value that has no associated curricula: the 'semester' value. So far, I haven't found any curricula that are labeled as a 'semester', is there one we should be expecting to be that? If not, should we have that filter value?

Reply:
> So maybe once the data is validated on the backend then we can remove the redundant checks for non-null values on the front-end and just have those values be required for each curriculum?

Yes, sounds good! I'll leave the question about the semester value up to Dani.

Reply:
Oh! Interesting find @TurnerRiley. I wil ask in the T&L channel and see what they think. I'm wondering if the new CSD collections will be more semester length.